modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkParameterGroupModelParser.java

```diff
@@ -20,26 +23,49 @@
 import org.mule.runtime.module.extension.internal.loader.parser.ParameterGroupModelParser;
 import org.mule.runtime.module.extension.internal.loader.parser.ParameterModelParser;
 
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
 
 /**
  * {@link ParameterGroupModelParser} implementation for Mule SDK
  *
  * @since 4.5.0
  */
-class MuleSdkParameterGroupModelParser implements ParameterGroupModelParser {
+class MuleSdkParameterGroupModelParser extends BaseMuleSdkExtensionModelParser implements ParameterGroupModelParser {
 
   private final List<ParameterModelParser> parameters;
+  private final Optional<ExclusiveOptionalDescriptor> exclusiveOptionals;
 
-  public MuleSdkParameterGroupModelParser(List<ComponentAst> parameters, TypeLoader typeLoader) {
-    this.parameters = doParserParameters(parameters, typeLoader);
+  public MuleSdkParameterGroupModelParser(ComponentAst parametersComponent, TypeLoader typeLoader) {
+    parameters = doParserParameters(parametersComponent, typeLoader);
+    exclusiveOptionals = doParseExclusiveOptionalDescriptorFromGroup(parametersComponent);
   }
 
-  private List<ParameterModelParser> doParserParameters(List<ComponentAst> parameters, TypeLoader typeLoader) {
-    return unmodifiableList(parameters.stream()
-        .map(p -> new MuleSdkParameterModelParserSdk(p, typeLoader))
-        .collect(toList()));
+  private List<ParameterModelParser> doParserParameters(ComponentAst parametersComponent, TypeLoader typeLoader) {
+    Stream<ParameterModelParser> parameterParsers = getChildren(parametersComponent, "parameter")
+        .map(p -> new MuleSdkParameterModelParserSdk(p, typeLoader));
+    Stream<ParameterModelParser> optionalParameterParsers = getChildren(parametersComponent, "optional-parameter")
+        .map(p -> new MuleSdkOptionalParameterModelParserSdk(p, typeLoader));
+
+    return unmodifiableList(concat(parameterParsers, optionalParameterParsers).collect(toList()));
```

Comment:
remove the unmodifiableList

Reply:
Note that the method [getParameterParsers](https://github.com/mulesoft/mule/blob/c89b364e48e2f806ce6374109fbefd94e67f327a/modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkParameterGroupModelParser.java#L83) will be returning the reference to the `parameters` field unwrapped.

I assumed this wrapping was done here originally to protect it from external modifications (because the toList collector does not provide any immutability guarantees). Since we don't need to modify it internally either, we could have it wrapped as unmodifiable at this point instead of doing it before returning it from the `getParametersParsers` method.

If you still think the wrapping is not necessary I would remove it. Or I could move it to the getter.